### PR TITLE
Feat/3729 settings available everywhere

### LINKED ISSUE #3729: Access Settings: Trezor disconnected 🔌
The option to modify Application and Coin Settings before the keys are shared (device is connected) with the application.

Design in [Figma](https://www.figma.com/file/BUMem1C7j4roQDyJf9yxQT/Exploring-Onboarding?node-id=862%3A536)

----
image
- Settings icon should be available on all Welcome screen variants in the top-right corner and behave the same as usual (dropdown with links to settings and guide). Guide icon should also be on the Welcome screen at the usual position.

----
image
- The layout of settings in case device is not connected in normal mode is the same as usual but without main Navigation Bar (so there is no way to navigate to Accounts or Dashboard)

- There is a new X button in settings (top right navigation item) that leads to last route before you open the Settings. 

- [nice to have] The `...` button should be removed from settings
  - Support - removed completely (it's in guide)
  - Show log - move to General tab as regular item
  - Debug menu - move to new tab

- If the device is not connected in normal or bootloader mode:
    - Device tab has only new banner and no possible settings
    - General settings has disabled labelling
    - Coins has all coins available (so the behaviour would be equivalent to situation, that you enable XRP with model T connected and then connect model One)

----
image
image

 - If the device is in bootloader mode:
   - Same as above except the device tab that has different banner and firmware update, wipe device and custom firmware is there
   - This is related to issue https://github.com/trezor/trezor-suite/issues/2542
   
 ----
image
- On top of that state of the Device tab in settings should be enhanced for the Remembered device
  - new banner added on top
  - cursor on all the options that are disabled should be `cursor: not-allowed` (crossed out circle)
  - there should be an explanation tooltip on disabled toggles
  
  
 - If you went to Settings from Welcome screen and then connect the device, you should stay in Settings just the Navigation bar should appear and so the state of the current settings tab
 - The same if you are in Settings with device and the device is disconnected, you should stay in Settings, but the Navigation bar disappear.

### PR DESCRIPTION
**Pull request by matejkriz**
fix #3729 (see the Description there)

### CODE REVIEW

File: packages/suite/src/components/guide/GuidePanel.tsx [hunk 1]
```diff
@@ -9,39 +11,72 @@ import {
...
     Feedback,
 } from '@guide-views';
+import { MODAL } from '@suite-actions/constants';
 
-const Wrapper = styled.div`
-    background: ${props => props.theme.BG_WHITE};
-    display: flex;
-    flex-direction: column;
+const GuideWrapper = styled.div<{ isModalOpen?: boolean }>`
+    z-index: 11;
```
Comment (matejkriz) [inline packages/suite/src/components/guide/GuidePanel.tsx]: What about to put it in Z_INDEX in `/packages/components/src/config/variables.ts`?

File: packages/suite/src/components/suite/WelcomeLayout/index.tsx [hunk 2]
```diff
@@ -84,50 +89,105 @@ const SettingsWrapper = styled.div`
...
+                <AnimatePresence>
+                    {!guideOpen && (
+                        <MotionWelcome
+                            initial={
+                                !firstRenderDone
+                                    ? {
+                                          width: welcomeBarWidth,
+                                      }
+                                    : { width: 0 }
+                            }
```
Comment (matejkriz) [inline packages/suite/src/components/suite/WelcomeLayout/index.tsx:137]: I believe that framer-motion could handle it without custom `firstRenderDone` state, but not sure.

Have you tried `initial={false}`? https://www.framer.com/docs/animation/##mount-animations

Comment (tomasklim) [reply packages/suite/src/components/suite/WelcomeLayout/index.tsx:137]: yes, I have. If you use `initial={false}`, guide opening animation is missing

Comment (matejkriz) [reply packages/suite/src/components/suite/WelcomeLayout/index.tsx:137]: 👌

File: packages/suite/src/hooks/guide/useGuideKeyboard.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import { KEYBOARD_CODE } from '@trezor/components';
+import * as guideActions from '@suite-actions/guideActions';
+import { useSelector, useActions } from '@suite-hooks';
+
+export const useGuideKeyboard = () => {
+    const { guideOpen, isModalOpen } = useSelector(state => ({
+        guideOpen: state.guide.open,
+        // 2 types of modals exist. 1. redux 'modal' based, 2. redux 'router' based
+        isModalOpen:
+            state.modal.context !== MODAL.CONTEXT_NONE || state.router.route?.isForegroundApp,
```
Comment (matejkriz) [inline packages/suite/src/hooks/guide/useGuideKeyboard.ts]: this selector is repeated on 3 places, maybe time for custom hook or util?

Comment (tomasklim) [reply packages/suite/src/hooks/guide/useGuideKeyboard.ts]: this selector is not final yet

Comment (tomasklim) [reply packages/suite/src/hooks/guide/useGuideKeyboard.ts]: more conditions to come

Comment (tomasklim) [reply packages/suite/src/hooks/guide/useGuideKeyboard.ts:13]: 5dd3f3dc8b87497e02bb5b906e08eacfe10e7534

Comment (tomasklim) [reply packages/suite/src/hooks/guide/useGuideKeyboard.ts]: 5dd3f3d

File: packages/suite/src/components/guide/GuidePanel.tsx (hunk 1 above)
Comment (tomasklim) [reply packages/suite/src/components/guide/GuidePanel.tsx]: db5ed42

File: packages/suite/src/components/suite/SuiteLayout/index.tsx [hunk 4]
```diff
@@ -11,6 +11,7 @@ import { DiscoveryProgress } from '@wallet-components';
 import NavigationBar from '../NavigationBar';
 import { useLayoutSize, useSelector, useDevice } from '@suite-hooks';
 import { MODAL } from '@suite-actions/constants';
```
Comment (matejkriz) [inline packages/suite/src/components/suite/SuiteLayout/index.tsx]: @tomasklim 'MODAL' is declared but its value is never read

Comment (tomasklim) [reply packages/suite/src/components/suite/SuiteLayout/index.tsx]: 00d4fa7

File: packages/suite/src/support/messages.ts [hunk 5]
```diff
@@ -2892,6 +2892,35 @@ export default defineMessages({
...
+        id: 'TR_SETTINGS_DEVICE_BANNER_TITLE_BOOTLOADER',
+    },
+    TR_SETTINGS_DEVICE_BANNER_TITLE_REMEMBERED: {
+        defaultMessage: 'Connect device to change device settings',
+        id: 'TR_SETTINGS_DEVICE_BANNER_TITLE_REMEMBERED',
+    },
+    TR_SETTINGS_DEVICE_BANNER_DESCRIPTION_DISCONNECTED: {
+        defaultMessage: 'TODO',
+        id: 'TR_SETTINGS_DEVICE_BANNER_DESCRIPTION_DISCONNECTED',
+    },
```
Comment (tomasklim) [inline packages/suite/src/support/messages.ts]: `TODO`, not used

File: packages/suite/src/support/messages.ts [hunk 6]
```diff
@@ -2892,6 +2892,35 @@ export default defineMessages({
...
+        id: 'TR_SETTINGS_DEVICE_BANNER_DESCRIPTION_BOOTLOADER_MODEL_1',
+    },
+    TR_SETTINGS_DEVICE_BANNER_DESCRIPTION_BOOTLOADER_MODEL_2: {
+        defaultMessage: 'Reconnect the device without touching the screen to access them.',
+        id: 'TR_SETTINGS_DEVICE_BANNER_DESCRIPTION_BOOTLOADER_MODEL_2',
+    },
+    TR_SETTINGS_DEVICE_BANNER_DESCRIPTION_REMEMBERED: {
+        defaultMessage: 'TODO',
+        id: 'TR_SETTINGS_DEVICE_BANNER_DESCRIPTION_REMEMBERED',
+    },
```
Comment (tomasklim) [inline packages/suite/src/support/messages.ts]: `TODO`, not used

File: packages/suite/src/support/messages.ts [hunk 7]
```diff
@@ -2892,6 +2892,35 @@ export default defineMessages({
...
+        defaultMessage: 'TODO',
+        id: 'TR_SETTINGS_DEVICE_BANNER_DESCRIPTION_DISCONNECTED',
+    },
+    TR_SETTINGS_DEVICE_BANNER_DESCRIPTION_BOOTLOADER_MODEL_1: {
+        defaultMessage:
+            'Reconnect the device and do not hold any buttons while connecting the cable to access them.',
+        id: 'TR_SETTINGS_DEVICE_BANNER_DESCRIPTION_BOOTLOADER_MODEL_1',
+    },
+    TR_SETTINGS_DEVICE_BANNER_DESCRIPTION_BOOTLOADER_MODEL_2: {
+        defaultMessage: 'Reconnect the device without touching the screen to access them.',
```
Comment (tomasklim) [inline packages/suite/src/support/messages.ts]: to access **them**. The word **them** sounds weird

File: packages/suite/src/support/messages.ts [hunk 8]
```diff
@@ -6853,6 +6853,22 @@ export default defineMessages({
...
+        id: 'TR_DEVICE_CONNECTED_BOOTLOADER_RECONNECT',
+        defaultMessage: 'In bootloader by mistake? Reconnect your device.',
+    },
+    TR_WIPE_OR_UPDATE: {
+        id: 'TR_WIPE_OR_UPDATE',
+        defaultMessage: 'Reset device or update firmware',
+    },
+    TR_WIPE_OR_UPDATE_DESCRIPTION: {
+        id: 'TR_WIPE_OR_UPDATE_DESCRIPTION',
+        defaultMessage: 'Go to device settigns',
```
Comment (tomasklim) [inline packages/suite/src/support/messages.ts]: `device settigns`

File: packages/suite/src/support/messages.ts (hunk 5 above)
Comment (matejkriz) [reply packages/suite/src/support/messages.ts]: https://github.com/trezor/trezor-suite/pull/4664/commits/559bc0401adf2d624a64f571bfe072ce8f346bc2

File: packages/suite/src/support/messages.ts (hunk 6 above)
Comment (matejkriz) [reply packages/suite/src/support/messages.ts]: https://github.com/trezor/trezor-suite/pull/4664/commits/559bc0401adf2d624a64f571bfe072ce8f346bc2

File: packages/suite/src/support/messages.ts (hunk 7 above)
Comment (matejkriz) [reply packages/suite/src/support/messages.ts]: https://github.com/trezor/trezor-suite/pull/4664/commits/559bc0401adf2d624a64f571bfe072ce8f346bc2

File: packages/suite/src/support/messages.ts (hunk 8 above)
Comment (matejkriz) [reply packages/suite/src/support/messages.ts]: https://github.com/trezor/trezor-suite/pull/4664/commits/559bc0401adf2d624a64f571bfe072ce8f346bc2

**Review by tomasklim — APPROVED** [review]
🙏🏽 ⛪️

**Pull request merged**